tests/test_init.py

```diff
@@ -17,542 +17,194 @@
+
+    @pytest.mark.parametrize("init", RPT_UNIF)
+    def test_rpt_interval_uniform(self, init, seed, n_wires, tol):
+        """Test that uniformly sampled initialization functions lie
+        in correct interval."""
+        p = init(n_layers=repeat, n_wires=n_wires, low=1, high=1, seed=seed)
+        p_mean = np.mean(p)
+        assert np.isclose(p_mean, 1, atol=tol)
+
+    @pytest.mark.parametrize("init", RPT_NRML)
+    def test_rpt_interval_normal(self, init, seed, n_wires, tol):
+        """Test that normal samples of initialization functions lie
+        in correct interval."""
+        p = init(n_layers=repeat, n_wires=n_wires, mean=1, std=0, seed=seed)
+        p_mean = np.mean(p)
+        assert np.isclose(p_mean, 1, atol=tol)
+
+
+class TestInitNotRepeated:
+    """Tests the initialization functions from the ``init`` module for
+    functions that have no ``repeat`` keyword argument."""
+
+    @pytest.mark.parametrize("init, shp", SHP)
+    def test_shape(self, init, shp, n_wires):
+        """Confirm that initialization functions without ``repeat``
+         returns an array with the correct shape."""
+        p = init(n_wires=n_wires, seed=0)
+        assert p.shape == shp
+
+    @pytest.mark.parametrize("init, shp", ALL_SHP)
+    def test_all_shape(self, init, shp, n_wires):
+        """Confirm that ``all`` initialization functions
+         return an array with the correct shape."""
+        p = init(n_wires=n_wires, seed=0)
+        shapes = [p_.shape for p_ in p]
+        assert shapes == shp
+
+    @pytest.mark.parametrize("init", COMBINED)
+    def test_same_output_for_same_seed(self, init, n_wires, seed, tol):
+        """Confirm that initialization function returns a deterministic output
+        for a fixed seed."""
+        p1 = init(n_wires=n_wires, seed=seed)
+        p2 = init(n_wires=n_wires, seed=seed)
+        assert np.allclose(p1, p2, atol=tol)
+
+    @pytest.mark.parametrize("init", COMBINED)
+    def test_diff_output_for_diff_seed(self, init, n_wires, seed, tol):
+        """Confirm that initialization function returns a different output for
+        different seeds."""
+        p1 = init(n_wires=n_wires, seed=seed)
+        p2 = init(n_wires=n_wires, seed=seed + 1)
+        assert not np.allclose(p1, p2, atol=tol)
+
+    @pytest.mark.parametrize("init", UNIF)
+    def test_interval_uniform(self, init, seed, n_wires, tol):
+        """Test that uniformly sampled initialization functions lie
+        in correct interval."""
+        p = init(n_wires=n_wires, low=1, high=1, seed=seed)
+        p_mean = np.mean(p)
+        assert np.isclose(p_mean, 1, atol=tol)
+
+    @pytest.mark.parametrize("init", NRML)
+    def test_interval_normal(self, init, seed, n_wires, tol):
+        """Test that normal samples of initialization functions lie
+        in correct interval."""
+        p = init(n_wires=n_wires, mean=1, std=0, seed=seed)
+        p_mean = np.mean(p)
+        assert np.isclose(p_mean, 1, atol=tol)
```

Comment:
Should we have integration tests in this file. That is, check that the generated parameters are accepted by the templates